Source/WebDriver/WebSocketServer.h

```diff
@@ -129,7 +129,10 @@ class WebSocketServer : public RefCountedAndCanMakeWeakPtr<WebSocketServer> {
     String getResourceName(const String& sessionId);
     String getWebSocketURL(const RefPtr<WebSocketListener>, const String& sessionId);
     String getSessionID(const String& resource);
-    void sendMessage(const String& session, const String& message);
+    void sendMessage(const String& sessionId, const String& message);
+    void sendMessage(WebSocketMessageHandler::Connection, const String& message);
```

Comment:
Can the methods that accept `WebSocketMessageHandler::Connection` as a parameter be made private to `WebSocketServer`?

From what I can tell, they're primarily used by the platform-specific (Soup) implementation. `WebDriverService` and `WebSocketServer` currently communicate using  both `sessionId` and `connection` but it seems that `sessionId` alone would suffice and allow for better encapsulation of the server logic.

Reply:
Agree with Yury's suggestion, except... As commented elsewhere, I prefer using `clientId` rather than `sessionId` for this purpose. For BiDi-only sessions that use the static session creation endpoint, they won't have a `sessionId` unless `session.new` succeeds. And even then, the client could send the `session.status` command which has no associated session.

Reply:
Moving the `Connection` to private is fine, I'll do it.

On the other hand, about `sessionId` vs `clientId`, that's a good point.

`session.status` is fine, because it replies through the completion handler directly, not passing through this method.

`session.new` on the other hand, will receive an async reply from the browser, but as you mentioned, its original `Connection` will not be mapped yet to any known `Session`.

Given most of the currently supported commands are handled through valid sessions, would it be ok to leave the `sessionId` in _this version_, and we add proper `clientId` support with static methods in a separate https://bugs.webkit.org/show_bug.cgi?id=281721?